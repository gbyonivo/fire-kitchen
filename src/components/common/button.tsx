import { Button as HeadlessButton } from "@headlessui/react";
import { Spinner } from "./spinner";

interface ButtonProps {
  children: React.ReactNode;
  className?: string;
  loading?: boolean;
  disabled?: boolean;
}

const disabledClassName = "bg-gray-600 data-open:bg-gray-600 text-gray-400";

const enabledClassName =
  "bg-gray-800 data-hover:bg-gray-700 data-open:bg-gray-800 text-white";

export function Button({
  children,
  className,
  loading,
  disabled,
}: ButtonProps) {
  return (
    <HeadlessButton
      className={`inline-flex items-center gap-2 rounded-md 
        px-3 py-1.5 text-sm/6 font-semibold  shadow-inner 
        shadow-white/10 focus:not-data-focus:outline-none data-focus:outline 
        data-focus:outline-white  ${
          disabled ? disabledClassName : enabledClassName
        } ${className}`}
      disabled={loading || disabled}
    >
      <div className="flex items-center gap-2">
        {loading && <Spinner size="small" />}
        {children}
      </div>
    </HeadlessButton>
  );
}
