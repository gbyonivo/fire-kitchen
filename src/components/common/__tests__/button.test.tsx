import { render } from "@testing-library/react";
import { Button } from "../button";

describe("Button", () => {
  test.each([
    [{ disabled: false, loading: false }],
    [{ disabled: true, loading: true }],
  ])("%s should match snapshot", ({ disabled, loading }) => {
    const onClick = jest.fn();
    const { container } = render(
      <Button onClick={onClick} disabled={disabled} loading={loading}>
        Test
      </Button>
    );
    expect(container).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
