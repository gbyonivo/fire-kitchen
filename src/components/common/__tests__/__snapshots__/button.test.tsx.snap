// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button { disabled: false, loading: false } should match snapshot 1`] = `
<div>
  <button
    class="inline-flex items-center gap-2 rounded-md 
        px-3 py-1.5 text-sm/6 font-semibold  shadow-inner 
        shadow-white/10 focus:not-data-focus:outline-none data-focus:outline 
        data-focus:outline-white  bg-gray-800 data-hover:bg-gray-700 data-open:bg-gray-800 text-white undefined"
    data-headlessui-state=""
    type="button"
  >
    <div
      class="flex items-center gap-2"
    >
      Test
    </div>
  </button>
</div>
`;

exports[`Button { disabled: true, loading: true } should match snapshot 1`] = `
<div>
  <button
    class="inline-flex items-center gap-2 rounded-md 
        px-3 py-1.5 text-sm/6 font-semibold  shadow-inner 
        shadow-white/10 focus:not-data-focus:outline-none data-focus:outline 
        data-focus:outline-white  bg-gray-600 data-open:bg-gray-600 text-gray-400 undefined"
    data-disabled=""
    data-headlessui-state="disabled"
    disabled=""
    type="button"
  >
    <div
      class="flex items-center gap-2"
    >
      <div>
        <svg
          aria-hidden="true"
          class="animate-spin text-red-500 undefined w-4 h-4"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.963 2.286a.75.75 0 0 0-1.071-.136 9.742 9.742 0 0 0-3.539 6.176 7.547 7.547 0 0 1-1.705-1.715.75.75 0 0 0-1.152-.082A9 9 0 1 0 15.68 4.534a7.46 7.46 0 0 1-2.717-2.248ZM15.75 14.25a3.75 3.75 0 1 1-7.313-1.172c.628.465 1.35.81 2.133 1a5.99 5.99 0 0 1 1.925-3.546 3.75 3.75 0 0 1 3.255 3.718Z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      Test
    </div>
  </button>
</div>
`;
